vj4/model/domain.py

```diff
@@ -214,6 +217,21 @@ def get_multi_user(*, fields=None, **kwargs):
   return result
 
 
+def get_join_settings(ddoc):
+  if 'join' not in ddoc:
+    return None
+  join_settings = ddoc['join']
+  if join_settings is None:
+    return None
+  if join_settings['method'] == constant.domain.JOIN_METHOD_NONE:
+    return None
+  if join_settings['role'] not in ddoc['roles']:
+    return None
+  if join_settings['expire'] != None and join_settings['expire'] < datetime.datetime.utcnow():
```

Comment:
`if join_settings['expire'] and join_settings['expire'] <`...

Reply:
they are not equal?